File: backend/app/schema/request.py
from typing import Optional

from pydantic import BaseModel, Field


class UpdateMemberInfoRequest(BaseModel):
    nickname: Optional[str] = None
    mc_name: Optional[str] = Field(None, alias="mcName")
    real_name: Optional[str] = Field(None, alias="realName")
    student_id: Optional[str] = Field(None, alias="studentID")
    college: Optional[str] = None
    school: Optional[str] = None
    major: Optional[str] = None
    grade: Optional[int] = None
    class_index: Optional[int] = Field(None, alias="classIndex")


class AddMemberRequest(BaseModel):
    qq_id: str = Field(..., alias="QQID")
    nickname: str
    password: str
    mc_name: Optional[str] = Field(None, alias="mcName")
    real_name: str = Field(..., alias="realName")
    student_id: Optional[str] = Field(None, alias="studentID")
    college: str
    school: Optional[str]
    major: Optional[str]
    grade: Optional[int]
    class_index: Optional[int] = Field(None, alias="classIndex")

File: frontend/src/schema/request.ts
export interface UpdateMemberInfoRequest {
  nickname?: string
  mcName?: string
  realName?: string
  studentID?: string
  college?: string
  school?: string
  major?: string
  grade?: number
  classIndex?: number
}

export interface AddMemberRequest {
  QQID: string
  nickname: string
  password: string
  mcName?: string
  realName: string
  studentID?: string
  college: string
  school?: string
  major?: string
  grade?: number
  classIndex?: number
}
